functions/__tests__/functions.test.ts

```diff
@@ -69,122 +77,147 @@ describe('extension', () => {
-    )
-
-    const documentChange = firebaseMock.makeChange(
-      beforeSnapshot,
-      afterSnapshot
-    )
-
-    const callResult = await mockExport(documentChange, {
-      resource: {
-        name: 'test',
-      },
-    })
-
-    expect(callResult).toBeUndefined()
     expect(mockConsoleLog).toBeCalledWith(
       'Initializing extension with configuration',
       config
     )
-    expect(mockConsoleLog).toBeCalledWith(
-      'Started execution of extension with configuration',
-      config
-    )
-    expect(mockConsoleLog).toBeCalledWith('Completed execution of extension')
-    expect(mockConsoleInfo).toBeCalledWith(
-      `Creating new document ${
-        afterSnapshot.id as string
-      } in MeiliSearch index ${defaultEnvironment.MEILISEARCH_INDEX_NAME}`,
-      { ...afterSnapshot.data() }
-    )
-    expect(mockedAddDocuments).toHaveBeenCalledWith([defaultDocument])
   })
 
-  test('function runs with updated data', async () => {
-    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
-      { foo: 'bar' },
-      'collection/doc'
-    )
-    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
-      defaultDocument,
-      'collection/doc'
-    )
-
-    const documentChange = firebaseMock.makeChange(
-      beforeSnapshot,
-      afterSnapshot
+  test('meilisearch getOrCreateIndex', () => {
+    expect(mockedGetOrCreateIndex).toHaveBeenCalledWith(
+      defaultEnvironment.MEILISEARCH_INDEX_NAME
     )
-
-    const callResult = await mockExport(documentChange, {
-      resource: {
-        name: 'test',
-      },
-    })
-
-    expect(callResult).toBeUndefined()
     expect(mockConsoleLog).toBeCalledWith(
       'Initializing extension with configuration',
       config
     )
+  })
+
+  test('meilisearch updateSearchableAttributes', () => {
+    expect(mockedUpdateSearchableAttributes).toHaveBeenCalledWith([
+      defaultEnvironment.SEARCHABLE_FIELDS,
+    ])
     expect(mockConsoleLog).toBeCalledWith(
-      'Started execution of extension with configuration',
+      'Initializing extension with configuration',
       config
     )
-    expect(mockConsoleInfo).toBeCalledWith(
-      `Updating document ${afterSnapshot.id as string} in MeiliSearch index ${
-        defaultEnvironment.MEILISEARCH_INDEX_NAME
-      }`,
-      { ...afterSnapshot.data() }
-    )
-    expect(mockConsoleLog).toBeCalledWith('Completed execution of extension')
-    expect(mockedUpdateDocuments).toHaveBeenCalledWith([defaultDocument])
   })
 
-  test('functions runs with deleted data', async () => {
-    const beforeSnapshot = { ...defaultDocument }
-    const afterSnapshot = { ...defaultDocument, exists: false }
+  describe('functions.indexingWorker', () => {
+    firebaseMock = firebaseFunctionsTestInit()
 
-    const documentChange = firebaseMock.makeChange(
-      beforeSnapshot,
-      afterSnapshot
-    )
+    // eslint-disable-next-line @typescript-eslint/require-await
+    beforeEach(async () => {
```

Comment:
Is there a reason why your beforeEach is async if there are no await on an async functions in it?

Reply:
No I put it here because the test was async too but actually it works without. I remote it